File: prototype/java/starcraft/src/main/java/lines/service/usecase/Marine.java
package lines.service.usecase;

import lines.model.Unit;
import lines.model.UnitAction;

public class Marine implements Unit {

    private final UnitAction unitAction;
    private int accumulatedMineral;

    public Marine(UnitAction unitAction){
        this.unitAction = unitAction;
    }

    @Override
    public int GerMineralCapacity() {
        return accumulatedMineral;
    }

    @Override
    public void SetMineralCapacity(int accumulatedMineral) {
        this.accumulatedMineral = accumulatedMineral;
    }

    @Override
    public void Harvest() {
        this.unitAction.Harvest();
    }

    @Override
    public void Attack() {
        this.unitAction.Attack();
    }

    @Override
    public void Building() {
        this.unitAction.Building();
    }
}

File: prototype/kotlin/starcraft/src/main/kotlin/lines/service/usecase/Marine.kt
package lines.service.usecase

import lines.model.Unit
import lines.model.UnitAction

class Marine constructor(val unitAction: UnitAction, var mineralCapacity: Int) : Unit {
    override fun GetMineralCapacity(): Int {
        return this.mineralCapacity
    }

    override fun SetMineralCapacity(accumulateMineral: Int) {
        this.mineralCapacity = accumulateMineral
    }

    override fun Harvest() {
        this.unitAction.Harvest()
    }

    override fun Attack() {
        this.unitAction.Attack()
    }

    override fun Building() {
        this.unitAction.Building()
    }
}